### PR TITLE
Document table redirection concept for the lakehouse connectors

### PR DESCRIPTION
**Pull request by findinpath**

### CODE REVIEW

File: docs/src/main/sphinx/connector/delta-lake.rst
Comment (alexjo2144) [inline docs/src/main/sphinx/connector/delta-lake.rst:349]: Phrasing nit, but I defer to the docs team. I find it clearer to put the situation you're addressing at the beginning of the sentence rather than at the end. 

```suggestion
When dealing with a non Delta Lake table, the connector supports redirecting queries to :doc:`/connector/hive`
using the configuration setting ``delta.hive-catalog-name``.
```

File: docs/src/main/sphinx/connector/table-redirection.fragment
Comment (mosabua) [inline docs/src/main/sphinx/connector/table-redirection.fragment:1]: "lakehouse connectors" is not a concept that we have defined in the docs anywhere.. so you need to enumerate them or whatever

File: docs/src/main/sphinx/connector/hive.rst
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:715]: The connector supports redirection from Hive tables to Iceberg and Delta Lake tables with the following catalog configuration properties:

File: docs/src/main/sphinx/connector/iceberg.rst
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:212]: Same as above but as sentence

File: docs/src/main/sphinx/connector/table-redirection.fragment
Comment (mosabua) [reply docs/src/main/sphinx/connector/table-redirection.fragment:1]: avoid "it is quite common" and other fillers

File: docs/src/main/sphinx/connector/table-redirection.fragment
Comment (mosabua) [inline docs/src/main/sphinx/connector/table-redirection.fragment:7]: Redirection also works without preselection right? So why bring it up?

File: docs/src/main/sphinx/connector/table-redirection.fragment
Comment (mosabua) [inline docs/src/main/sphinx/connector/table-redirection.fragment:18]: Avoid comments in the code snippet .. just break it up into multiple snippets .

**Review by mosabua — CHANGES_REQUESTED** [review]
Needs a bunch of rewording and rejigging.. feel free to ping me to work on this together

File: docs/src/main/sphinx/connector/table-redirection.fragment
Comment (colebow) [inline docs/src/main/sphinx/connector/table-redirection.fragment:5]:
```suggestion
Therefore, a metastore database can hold a variety of tables
with different table formats.
```

File: docs/src/main/sphinx/connector/table-redirection.fragment
Comment (colebow) [inline docs/src/main/sphinx/connector/table-redirection.fragment:13]:
```suggestion
As a concrete example, let's use the following
```

File: docs/src/main/sphinx/connector/table-redirection.fragment
Comment (findinpath) [reply docs/src/main/sphinx/connector/table-redirection.fragment]: Good point. I added a corresponding sample snippet for this use case.

File: docs/src/main/sphinx/connector/table-redirection.fragment
Comment (colebow) [inline docs/src/main/sphinx/connector/table-redirection.fragment:3]:
```suggestion
(for example, :doc:`/connector/hive`, :doc:`/connector/iceberg`, and :doc:`/connector/delta-lake`),
the metastore (Hive metastore service, `AWS Glue Data Catalog <https://aws.amazon.com/glue/>`_)
```

**Review by colebow — APPROVED** [review]

File: docs/src/main/sphinx/connector/table-redirection.fragment
Comment (phd3) [inline docs/src/main/sphinx/connector/table-redirection.fragment:5]: This is the motivation for initial usecases, but not the requirement. IMO we can swap this with the second paragraph. (i.e. first introduce the concept of redirection, and then say that it's currently implemented for connectors depending on metastore service)

File: docs/src/main/sphinx/connector/table-redirection.fragment
Comment (phd3) [inline docs/src/main/sphinx/connector/table-redirection.fragment:8]: may be mention

```
based on the format of the table and catalog configuration
```

**Review by phd3 — COMMENTED** [review]
Looks good generally

**Review by phd3 — APPROVED** [review]
will merge unless @mosabua would like to take another look

**Review by mosabua — APPROVED** [review]
Looks good!

**Pull request merged**
